gandalf.js

```diff
@@ -3,59 +3,58 @@
  *
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.